skrf/media/media.py

```diff
@@ -643,14 +643,27 @@ def resistor(self, R: NumberLike, *args, **kwargs) -> Network:
         capacitor
         inductor
         """
-        result = self.match(nports=2, **kwargs)
-        y = np.zeros(shape=result.s.shape, dtype=complex)
+        s_def = kwargs.pop('s_def', S_DEF_DEFAULT)
+        result = self.match(nports=2, s_def=S_DEF_DEFAULT, **kwargs)
```

Comment:
I doubt we will change `S_DEF_DEFAULT` but to make sure it works if we do, this should be `s_def='power'`. Same in other classes.

Reply:
All S-Parameters that are only accurate under the definition of 'power' have been changed to explicitly `s_def='power'`.

Reply:
`kwargs.pop` should still have `S_DEF_DEFAULT` to make sure the correct `s_def` is returned if `S_DEF_DEFAULT` is not power.

Reply:
Modified as suggested.